# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/vr/ipc/VRLayerParent.cpp
# Commit: e1b0906509ef
# Full Hash: e1b0906509efa6433978b9f980ec639b7eafbc6d
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Fix non-webgl CI tests. r=handyman
#   
#   * Revert some partial webgl+oop+vr code.
#   * More missing FuncScope.
#   * Fix compile errors.
# ==============================================================================

diff -r 7e2a2b1b416f -r e1b0906509ef gfx/vr/ipc/VRLayerParent.cpp
--- a/gfx/vr/ipc/VRLayerParent.cpp	Wed Jan 08 22:19:23 2020 +0000
+++ b/gfx/vr/ipc/VRLayerParent.cpp	Wed Jan 08 22:19:26 2020 +0000
@@ -5,12 +5,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "VRLayerParent.h"
-
-#include "mozilla/dom/WebGLParent.h"
+#include "mozilla/Unused.h"
 #include "mozilla/layers/CompositorThread.h"
-#include "mozilla/Unused.h"
-#include "SharedSurface.h"
-#include "SurfaceTypes.h"
 
 namespace mozilla {
 using namespace layers;
@@ -43,67 +39,15 @@
 }
 
 mozilla::ipc::IPCResult VRLayerParent::RecvSubmitFrame(
-    mozilla::dom::PWebGLParent* aPWebGLParent, const uint64_t& aFrameId,
-    const uint64_t& aLastFrameId, const gfx::Rect& aLeftEyeRect,
-    const gfx::Rect& aRightEyeRect) {
-  // Keep the SharedSurfaceTextureClient alive long enough for
-  // 1 extra frame, accomodating overlapped asynchronous rendering.
-  mLastFrameTexture = mThisFrameTexture;
-
-  auto webGLParent = static_cast<mozilla::dom::WebGLParent*>(aPWebGLParent);
-  if (!webGLParent) {
-    return IPC_OK();
-  }
-
-#if defined(MOZ_WIDGET_ANDROID)
-  /**
-   * Do not blit WebGL to a SurfaceTexture until the last submitted frame is
-   * already processed and the new frame poses are ready. SurfaceTextures need
-   * to be released in the VR render thread in order to allow to be used again
-   * in the WebGLContext GLScreenBuffer producer. Not doing so causes some
-   * freezes, crashes or other undefined behaviour.
-   * DLP: TODO: Umm... not even the same process as before so...
-   */
-  if (!mThisFrameTexture || aLastFrameId == mLastSubmittedFrameId) {
-    mThisFrameTexture = webGLParent->GetVRFrame();
-  }
-#else
-  mThisFrameTexture = webGLParent->GetVRFrame();
-#endif  // defined(MOZ_WIDGET_ANDROID)
-
-  if (!mThisFrameTexture) {
-    return IPC_OK();
-  }
-
-  // TODO: I killed all of the syncObject stuff rather than move it here
-  // cause my current understanding is that it isn't needed here.
-
-  gl::SharedSurface* surf = mThisFrameTexture->Surf();
-  if (surf->mType == gl::SharedSurfaceType::Basic) {
-    gfxCriticalError() << "SharedSurfaceType::Basic not supported for WebVR";
-    return IPC_OK();
-  }
-
-  layers::SurfaceDescriptor desc;
-  if (!surf->ToSurfaceDescriptor(&desc)) {
-    gfxCriticalError() << "SharedSurface::ToSurfaceDescriptor failed in "
-                          "VRLayerParent::SubmitFrame";
-    return IPC_OK();
-  }
-
+    const layers::SurfaceDescriptor& aTexture, const uint64_t& aFrameId,
+    const gfx::Rect& aLeftEyeRect, const gfx::Rect& aRightEyeRect) {
   if (mVRDisplayID) {
     VRManager* vm = VRManager::Get();
-    vm->SubmitFrame(this, desc, aFrameId, aLeftEyeRect, aRightEyeRect);
+    vm->SubmitFrame(this, aTexture, aFrameId, aLeftEyeRect, aRightEyeRect);
   }
 
   return IPC_OK();
 }
 
-mozilla::ipc::IPCResult VRLayerParent::RecvClearSurfaces() {
-  mThisFrameTexture = nullptr;
-  mLastFrameTexture = nullptr;
-  return IPC_OK();
-}
-
 }  // namespace gfx
 }  // namespace mozilla